js/src/config/ProtectedRoute.jsx

```diff
@@ -1,39 +1,46 @@
 import { useState, useEffect } from 'react';
 import { Navigate } from 'react-router-dom';
 import { fetchAuthSession } from 'aws-amplify/auth';
+import { useAuthStore } from '../store/useStore';
+import { defineUserAbility } from '../ability/defineAbility';
 import { LoadingSpinner } from '../components';
 
-const ProtectedRoute = ({ children }) => {
-  const [isAuthenticated, setIsAuthenticated] = useState(false);
+const ProtectedRoute = ({ children, role }) => {
+  const { user, setUser, ability, setAbility } = useAuthStore();
   const [loading, setLoading] = useState(true);
 
   useEffect(() => {
     const checkAuth = async () => {
       try {
         const session = await fetchAuthSession();
-
         if (session?.tokens?.idToken) {
-          setIsAuthenticated(true);
+          const userRole = session.tokens.idToken.payload['custom:role'].toLowerCase(); // Convert to lowercase
```

Comment:
remove these comments.
Add comments if its informing about the logic of function

Reply:
removed the comments